pmd-swift/src/main/antlr4/net/sourceforge/pmd/lang/swift/ast/Swift.g4

```diff
@@ -297,6 +298,10 @@ typealiasHead : attributes? accessLevelModifier? 'typealias' typealiasName gener
 typealiasName : identifier  ;
 typealiasAssignment : '=' sType  ;
 
+// GRAMMAR OF A MACRO DECLARATION
+
+macroDeclaration: '#' identifier ;
```

Comment:
This actually looks a bit incomplete: that would mean, that a macro only consists of a name, but it can have parameters, similar like a function call, like in your example

```
#fontLiteral(name: "SF Mono", size: 14, weight: .regular)
```

Here's the actual grammar documented: https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md#macro-expansion

Maybe it should be more along the lines (not tested):

```
macroExpansionExpression: '#' identifier genericArgumentClause? functionCallArgumentClause?
```

And note: this is an expression, not a declaration. So it should not be added into the "declaration" production, but somewhere in "primaryExpression" (line 588)...


Macro declarations look differently: https://github.com/apple/swift-evolution/blob/main/proposals/0382-expression-macros.md#macro-declarations

Reply:
CI is fine (now fails on pmd-dist